src/infrastructure/BaseService.ts

```diff
@@ -11,23 +11,24 @@ interface BaseModelOptions {
   rejectUnauthorized?: boolean;
 }
 
+export type BaseModelContructorOptions =
```

Comment:
Should these options be in their own directory ?

Reply:
It's a matter of taste. I personally, keep types in the same file as long as it doesn't affect readability

Reply:
I would keep them in if they are relatively small, but i feel like the types information will start growing pretty quickly. It may be easier to read if they were all located together instead of being embedded in the file.